pandas/core/generic.py

```diff
@@ -10845,12 +10843,10 @@ def ewm(
         adjust: bool_t = True,
         ignore_na: bool_t = False,
         axis: Axis = 0,
-        times: str | np.ndarray | FrameOrSeries | None = None,
+        times: str | np.ndarray | NDFrame | None = None,
```

Comment:
Based on your comment about inheritance, I replaced all `FrameOrSeries` (can be any `NDFrame`) with `NDFrame` in public APIs.

Reply:
yes but that was for TypeVars, where FrameOrSeries is used and a TypeVar is not needed (i.e. not bound because only used once in the function signature or not part of a Generic class), we should replace with `DataFrame | Series` in the first instance? and if that gives errors maybe use NDFrame where it makes sense, e.g. core.generic?

Reply:
I think I'm missing two points here:

1. unneeded/invalid TypeVars (for me that includes TypeVars that are not bound) and
2. in which cases we want to allow other instances of `NDFrame` (my understanding was that a conservative replacement of not needed/invalid  `FrameOrSeries` is `NDFrame` but to prefer `DataFrame | Series` in private APIs when possible).

Reply:
so just to be clear and hopefully summarise the conversation so far:

pyright reports some errors (reportInvalidTypeVarUse) for some uses of the FrameOrSeries alias.

FrameOrSeries is defined as `TypeVar("FrameOrSeries", bound="NDFrame")`

so to fix the pyright errors we need to replace the FrameOrSeries typevar with a class or union of classes.

there are two options `FrameOrSeries` -> `DataFrame | Series` or `FrameOrSeries` -> `NDFrame`

if we do `FrameOrSeries` -> `NDFrame` then there is effectively no change to the types and mypy should not report any errors.

if we do `FrameOrSeries` -> `DataFrame | Series` then there will be some mypy errors.

The second option is the preferred option to be consistent with the recently removed `FrameOrSeriesUnion`

Reply:
okay, got it :) thank you for summarizing it!